qemu/tests/luks_image_over_qsd.py

```diff
@@ -0,0 +1,44 @@
+import socket
+
+from provider.qsd import QsdDaemonDev
+from virttest import error_context, env_process
+from virttest import qemu_storage
+
+
+# This decorator makes the test function aware of context strings
+@error_context.context_aware
+def run(test, params, env):
+    """
+        Export a luks image via QSD with NBD inet.
+        1. Create a luks image.
+        2. Start a QSD daemon, and export the luks image with NBD inet.
+        3. Check the info of the image over NBD.
+        4. Boot up a guest with the exported image as a data disk.
+        5. Check there is the data disk in the guest.
+        6. Connect to the QMP sock
+        6.1 Query the block exports
+        6.2 luks key management.
+        :param test: VT test object.
+        :param params: Dictionary with the test parameters.
+        :param env: Dictionary with test environment.
+        """
+    qsd = QsdDaemonDev("qsd1", params)
+    qsd.start_daemon()
+
+    nbd_image_tag = params['nbd_image_tag']
+    nbd_image_params = params.object_params(nbd_image_tag)
+    localhost = socket.gethostname()
+    nbd_image_params['nbd_server'] = localhost if localhost else 'localhost'
+    qemu_img = qemu_storage.QemuImg(nbd_image_params, None, nbd_image_tag)
+    qemu_img.info()
+
+    params["start_vm"] = 'yes'
+
+    login_timeout = int(params.get("login_timeout", 360))
```

Comment:
there is a `params.get_numeric()`
eg: `mem = params.get_numeric("mem")`

Reply:
Updated. Thanks.